lms/djangoapps/grades/models.py

```diff
@@ -257,8 +257,14 @@ def save_grade(cls, **kwargs):
                     usage_key=usage_key,
                     defaults=kwargs,
                 )
+            log.info("persistent subsection grade successfully saved: {0}".format(unicode(grade)))
         except IntegrityError:
             cls.update_grade(user_id=user_id, usage_key=usage_key, **kwargs)
+            log.info("integrity error trying to save grade for user {0}, usage key {1}, defaults {2}".format(
```

Comment:
Sounds like a plan to me. I'd suggest adding a logged warning in the exception handler, so we can have some data about how often this occurs when we get to grooming the celery task ticket.